app/assets/javascripts/discourse/app/components/post-text-selection.gjs

```diff
@@ -97,113 +81,8 @@ export default class PostTextSelection extends Component {
   }
 
   @bind
-  async hideToolbar() {
-    if (this.preventClose) {
-      return;
-    }
-    this.args.quoteState.clear();
-    await this.menuInstance?.close();
-  }
-
-  async selectionChanged(options = {}) {
```

Comment:
most of this code has been extracted in two functions:
- computeQuoteState
- computeSupportsFastEdit

It's also been generally simplified by relying on a simpler technique to know if our text selection is valid or not (common ancestor being .cooked)